src/braket/pennylane_plugin/braket_device.py

```diff
@@ -84,8 +82,7 @@ def __init__(
         shots: int,
         **run_kwargs,
     ):
-        # `shots` cannot be set to 0, but is ignored anyways
-        super().__init__(wires, shots or 1, analytic=shots == 0)
+        super().__init__(wires, shots=shots)
```

Comment:
```suggestion
        super().__init__(wires, shots=shots or None)
```
@licedric, this should be sufficient (although maybe hacky) to allow users to set `shots=0` and still get analytic mode.

Reply:
I would be interesting in learning on what the experience is like for other Pennylane integrations. Do we bias towards porting the experience for the dependencies we integrate with, or does Pennylane prefer that the users program with Pennylane specific parameters? Ultimately, pennylane users would want to follow a specific convention, that is either they only refer to the parameter convention for the specific integration (which they might anyways have to do for parameters specific to the integrated device) or, they follow the pennylane style parameters.

Reply:
@kshitijc, do you see a problem with allowing both the PennyLane and the Braket convention?

Reply:
Trying to support both usually tends to cause issues. However, I think supporting both shots=0/None could probably be less jarring for customers using both PL and Braket. Unless, Pennylane plans on implementing some behavior for shots=0 in the future, I'd be okay with supporting both for now.

Reply:
Things are complicated by the SDK [already having](https://github.com/aws/amazon-braket-sdk-python/blob/main/src/braket/aws/aws_device.py#L83) a default value of `None`, which is then used to select the device-specific default value.

Reply:
I suppose, stepping back a bit, this is more an issue with having a single PennyLane device for using the remote Braket devices (`BraketAwsQubitDevice`)? By having separate devices per remote device, sensible defaults could be chosen per remote device.